test/integration/util/util.go

```diff
@@ -43,17 +44,38 @@ type OcRunner struct {
 }
 
 func runCommand(command string, commandPath string) (stdOut string, stdErr string, exitCode int) {
+	return runCommandWithTimeout(command, commandPath, 0)
+}
+
+func runCommandWithTimeout(command string, commandPath string, timeout int) (stdOut string, stdErr string, exitCode int) {
+	var ctx context.Context
+	var cancel context.CancelFunc
+
 	commandArr := utilCmd.SplitCmdString(command)
 	path, _ := filepath.Abs(commandPath)
-	cmd := exec.Command(path, commandArr...)
+
+	if timeout > 0 {
+		ctx, cancel = context.WithTimeout(context.Background(), time.Second*time.Duration(timeout))
+	} else {
+		ctx, cancel = context.WithCancel(context.Background())
+	}
+	defer cancel()
```

Comment:
@agajdosi This is an interesting piece of code. There are different ways in go to achieve this. So wondering why you choose this. Also, I found https://medium.com/@vCabbage/go-timeout-commands-with-os-exec-commandcontext-ba0c861ed738

Reply:
Thanks for link, @LalatenduMohanty, if I remember correctly, I have seen the article as part of searching how to achieve the functionality. The context solution seemed to me to be more understandable, so I have used it instead of the second solution with the usage of goroutine. But no hard feelings over it, just checked and it looks like we do not use the context package anywhere else in Minishift, might be an argument against its usage.

Reply:
@agajdosi  I think the use of context is the right to do it.